src/allmydata/node.py

```diff
@@ -707,6 +709,7 @@ def create_tub(tub_options, default_connection_handlers, foolscap_connection_han
         the new Tub via `Tub.setOption`
     """
     tub = Tub(**kwargs)
+    support_foolscap_and_https(tub)
```

Comment:
The `Tub` is created immediately above so the interface to `support_foolscap_and_https` could be more like:

```
def support_foolscap_and_https(**kw) -> Tub:
    ...
```

(or the 5x longer version with correct type annotations)

Then the mutation of the tub is all hidden inside `support_foolscap_and_https` which seems better to me.